### DISCUSSION TITLE
Proper way to upgrade LXDWare?

### DISCUSSION THREAD

**Discussion by rkelleyrtp** [lxdware/lxd-dashboard#68] [General]
Maybe I missed it somewhere, but what is the proper way to upgrade LXDWare?  I am currently running LXDWare v3.1.0 and see there is a newer release (3.6).  I don't want to loose my current DB or UI configuration.

**Comment by matthewalanpenning** [comment]
That is a great question. You should be able to follow the same installation instructions for any 3.x version of the software If you have made any changes to the UI's code you should merge them into your copy of the source code. The database code has been written to update any schema changes automatically without losing your records. Of course it never hurts to make a backup of the db as a precaution.

**Comment by rkelleyrtp** [comment]
Thanks Matthew.  Perhaps a small blog or write-up on upgrading LXDWare would be valuable?  In the end, it was very straightforward.  
Here is a quick how-to for others:

Some notes beforehand:
- I am running LXDWare v3.1.0 inside an Alpine container 
- I want to upgrade to LXDWare v3.6.0
- The name of my LXDWare container is "lxd-dashboard"
- The HTML root directory for lxdware in this container is /www
- The LXDWare DB directory in this container is /var/lxdware/data

Process to update LXDWare in my Alpine container (adjust for your install as appropriate):
- Take a snapshot of the container running LXD (lxc snapshot lxd-dashboard)
- Login to the container running LXDWare
- cd /tmp 
- wget https://github.com/lxdware/lxd-dashboard/archive/refs/tags/v3.6.0.tar.gz
- tar xvzf v3.6.0.tar.gz
- cp /var/lxdware/data /var/lxdware/backups/<date_of_backup>
- cd /www
- mv lxd-dashboard lxd-dashboard_old
- cp -a /tmp/lxd-dashboard-3.6.0/lxd-dashboard /www/lxd-dashboard
- rm /tmp/lxd-dashboard-3.6.0* 
- reboot

Overall, it took about 3mins to update LXDWare.

**Reply by Wizlonex** [reply]
Works for upgrading to 3.7 too.  Excellent tutorial.  In my case, my instance was at /var/www/html/lxd-dashboard vs /www/lxd-dashboard, so I had to "cd /var/www/html" instead, and I changed the version number from 3.6.x to 3.7.x, but everything else worked like a charm.    THANK YOU.
